### DISCUSSION TITLE
TimeScale all Tweens

### DISCUSSION THREAD

**Discussion by peyman1959** [KyryloKuzyk/PrimeTween#127] [FAQ] [answered]
Hi
How can I set the Time Scale for all tweens instantly?
Dotween has a Dotween.timeScale that dose "Global timeScale applied to all tweens, both regular and independent."
is there somthing like that ? setting time scale withouth having access to sequences?

**Comment by KyryloKuzyk** [comment]
Hey! 

PrimeTween doesn't have this feature yet. It would be very easy to add it, but I would like to first understand the potential use case.
Would it be possible for you to set the `UnityEngine.Time.timeScale` instead? If not, can you please elaborate on why?

**Reply by peyman1959** [reply]
Changing UnityEngine.Time.timeScale would affect everything: the update loop, physics, etc. I only want to control the speed of my tweens.
DoTween actually did something pretty neat about controlling timescale like timescale-independent tweens.
In my scenario, I want the player to be able to skip animations by speeding them up when the player clicks; however, I don't want other stuff to speed up too.

**Reply by KyryloKuzyk** [reply]
@peyman1959 I see, that makes sense, I'll add this to my roadmap.

**Comment by KyryloKuzyk** [comment, marked as answer]
@peyman1959 
> In my scenario, I want the player to be able to skip animations by speeding them up when the player clicks; however, I don't want other stuff to speed up too.

I have been thinking about adding a new `PrimeTweenConfig.timeScale` property and found a few potential limitations. If a user wants to speed up only a few selected animations but leave others unchanged, the `PrimeTweenConfig.timeScale` will still not be enough. And I imagine that would be quite a common situation. So it seems to me that adding `PrimeTweenConfig.timeScale` would be limiting. 

Instead, I updated the implementation of the `TweenGroup` to also provide `timeScale` property and `TweenTimeScale()` method: https://github.com/KyryloKuzyk/PrimeTween/discussions/26#discussioncomment-7700985

To apply it to your use case, add all animation that you wish to speed up to the TweenGroup, then you can set the timeScale immediately or animate it for the whole group. What do you think?

```csharp
readonly TweenGroup group = new TweenGroup();

public void Test() {
    // Add animations to the group:
    group.Add(Tween.PositionX(transform, 10f, duration: 1f));
    group.Add(
        Sequence.Create()
            .ChainDelay(1f)
            .Chain(Tween.Rotation(transform, new Vector3(0f, 90f), duration: 1f))
    );

    // Set group timeScale immediately:
    group.timeScale = 0.5f;

    // Animate group timeScale:
    group.AnimateTimeScale(endValue: 0.5f, duration: 1f);

    // Stop or complete the group:
    group.Stop();
    group.Complete();
}
```

**Reply by peyman1959** [reply]
its good thank you
